airbyte-cdk/python/airbyte_cdk/sources/file_based/exceptions.py

```diff
@@ -14,6 +14,9 @@ class FileBasedSourceError(Enum):
     GLOB_PARSE_ERROR = (
         "Error parsing glob pattern. Please refer to the glob pattern rules at https://facelessuser.github.io/wcmatch/glob/#split."
     )
+    ENCODING_ERROR = (
+        "Error decoding file. Please check that the configured encoding matches the encoding of the file. Default encoding is UTF-8."
```

Comment:
is there anyway we can show the actual diff between the file and the configured encoding instead of only pointing the the default value?

Maybe the message can be set on the encoding error then set on the traced exception's message

Reply:
I think it's possible but not simple, since the error thrown doesn't tell what the encoding of the file is. It only complains about a missing BOM. We would need to run some kind of tool to guess the encoding of the file. What is more feasible is we can give the full message from DecodeError in the stacktrace, and tell what is the configured encoding is, rather than the default ...